soroban-sdk/src/env.rs

```diff
@@ -515,10 +527,27 @@ impl Env {
             .set_diagnostic_level(internal::DiagnosticLevel::Debug)
             .unwrap();
         env_impl.set_base_prng_seed([0; 32]).unwrap();
+
+        let auth_snapshot = Rc::new(RefCell::new(AuthSnapshot::default()));
+        let auth_snapshot_in_hook = auth_snapshot.clone();
+        env_impl
+            .set_top_contract_invocation_hook(Some(Rc::new(move |host, event| {
+                if let ContractInvocationEvent::Finish = event {
+                    let new_auths = host
+                        .get_authenticated_authorizations()
+                        // If an error occurs getting the authenticated authorizations
+                        // it means that no auth has occurred.
```

Comment:
I don't think error can ever occur here, because by definition we call this hook after invoking a contract and that guarantees that auth is present (even if empty). So if it occurs, then we likely have a bug in host.